src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -2041,4 +2157,22 @@ private static void FormatTrackedParameters(FactoryContext factoryContext, Strin
             errorMessage.AppendLine(FormattableString.Invariant($"{kv.Key,-19} | {kv.Value,-15}"));
         }
     }
+
+    // Due to cyclic references between Http.Extensions and
+    // Http.Results, we define our own instance of the `EmptyHttpResult`
+    // type here.
+    private sealed class EmptyHttpResult : IResult
```

Comment:
I'm with you. Since we're all pretty annoyed by this, I've filed https://github.com/dotnet/aspnetcore/issues/41330 to see if we could possibly remove the cyclic dependencies we have in our assemblies at the moment.